modules/apim_gateway_master/build.sh

```diff
@@ -0,0 +1,164 @@
+  cp ${product_binary} ${puppet_env}/modules/apim_gateway/files/
+}
+
+# Check if user has a WSO2 subscription
+while :
+do
+  read -p "Do you have a WSO2 subscription? (Y/n) "
+  echo
+  if [[ $REPLY =~ ^[Yy]$ ]] || [[ -z "$REPLY" ]]
+  then
+    if [[ ! -f ${install_path}/bin/update_linux ]]
+    then
+      echo "Update executable not found. Please download package for subscription users from website."
+      exit 1
+    else
+      break
+    fi
+  elif [[ $REPLY =~ ^[Nn]$ ]]
+  then
+    copy_to_agent "modified"
+    exit 0
+  else
+    echo "Invalid input provided."
+    sleep .5
+  fi
+done
+
+# Create updates directory if it doesn't exist
+if [[ ! -d ${distribution_path}/updates ]]
+then
+  mkdir ${distribution_path}/updates
+fi
+
+# Getting update status
+# 0 - first/last update successful
+# 1 - Error occurred in last update
+# 2 - In-place has been updated
+# 3 - conflicts encountered in last update
+status=0
+if [[ -f ${distribution_path}/updates/status ]]
+then
+  status=$(cat ${distribution_path}/updates/status)
+fi
+
+# Move into binaries of installation directory
+cd ${install_path}/bin
+
+# Run in-place update
+if [[ ${status} -eq 0 ]] || [[ ${status} -eq 1 ]] || [[ ${status} -eq 2 ]]
+then
+  ./update_linux --verbose 2>&1 | tee ${install_path}/bin/output.txt
+  update_status=${PIPESTATUS[0]}
+elif [[ ${status} -eq 3 ]]
+then
+  ./update_linux --verbose --continue 2>&1 | tee ${install_path}/bin/output.txt
+  update_status=${PIPESTATUS[0]}
+
+  # Handle user running update script without resolving conflicts
+  if [[ ${update_status} -eq 1 ]]
+  then
+    echo "Error occurred while attempting to resolve conflicts."
+    exit 1
+  fi
+else
+  echo "status file is invalid. Please delete or clear file content."
+  exit 1
+fi
+
+# Update status
+echo ${update_status} > ${distribution_path}/updates/status
+if [[ ${update_status} -eq 0 ]]
+then
+  echo
+  echo "Update completed successfully."
+  copy_to_agent "updated"
+elif [[ ${update_status} -eq 3 ]]
+then
+  echo "Conflicts encountered. Please resolve conflicts and run the update script again."
+else
+  echo "Update error occurred. Stopped with exit code ${update_status}"
+  exit ${update_status}
+fi
+
+# Get list of merged files
+if [[ ${update_status} -eq 0 ]] # If update is successful
+then
+  sed -n '/files./,/Successfully/p' ${install_path}/bin/output.txt > ${install_path}/bin/merged_files.txt
+elif [[ ${update_status} -eq 3 ]] # If conflicts were encountered during update
+then
+  sed -n '/files./,/Merging/p' ${install_path}/bin/output.txt > ${install_path}/bin/merged_files.txt
+fi
+
+if [[ -s ${install_path}/bin/merged_files.txt ]]
+then
+  sed -i '1d' ${install_path}/bin/merged_files.txt # Remove first line from file
+  sed -i '$ d' ${install_path}/bin/merged_files.txt # Remove last line from file
+
+  while read -r line; do
+    filepath=${line##*${product}-${product_version}/}
+    template_file=${puppet_env}/modules/is_master/templates/carbon-home/${filepath}.erb
```

Comment:
Shouldn't this be apim_gateway_master?

Reply:
Yes! I'll fix that and all the other build scripts immediately.

Reply:
Resolved https://github.com/wso2/puppet-apim/pull/86/commits/5dd25054bc378c8b256e4571b433b51eecc8d807